### PR TITLE
fix(Exchange): unsubscribe from unused currency pair.

### PR DESCRIPTION
**Pull request by bchrisarriola**
## Objective

Unsubscribe to rates for currency pairs that are different from the pair the user is trying to exchange in.

## Description

Prevents chatty/extraneous websocket messages where conversion rates are still being received from previous currency pairs. This can happen if the user was previously trying to exchange in a currency pair (e.g. "BTC-ETH") but then changed to a different pair (e.g. "BCH-ETH"). 

## How to Test

Pull in changes, go to create exchange flow, change currency pairs, and verify that you see the logged message: "Pair <old pair> is different from model pair <new pair>"

## Merge Checklist

- [X] The PR uses a title supported by [.changelogrc](https://github.com/blockchain/My-Wallet-V3-iOS/blob/dev/.changelogrc#L6...L69).
- [ ] Areas of technical debt are marked with a `// TICKET:` comment that includes a ticket number.
- [X] All unit tests pass.
- [ ] You have added unit tests.
- [ ] New files added are within the correct directory. (e.g. if a file is required for unit tests to compile, be sure it is added to the tests target.)

### CODE REVIEW

File: Blockchain/Homebrew/Exchange/Interactors/ExchangeCreateInteractor.swift [hunk 1]
```diff
@@ -90,8 +90,17 @@ extension ExchangeCreateInteractor: ExchangeCreateInput {
     func subscribeToConversions() {
         disposable = markets.conversions.subscribe(onNext: { [weak self] conversion in
             guard let this = self else { return }
-            guard let model = this.model, model.pair.stringRepresentation == conversion.quote.pair else {
-                Logger.shared.error("Pair returned from conversion is different from model pair")
+
+            guard let model = this.model else { return }
+
+            guard model.pair.stringRepresentation == conversion.quote.pair else {
```
Comment (woudini) [inline Blockchain/Homebrew/Exchange/Interactors/ExchangeCreateInteractor.swift]: I think the overall client logic will be cleaner/more intuitive if it rejected conversions whose trading pair didn't match the `model`'s and instead unsubscribed when `model.pair` changes. Thoughts?

Comment (bchrisarriola) [reply Blockchain/Homebrew/Exchange/Interactors/ExchangeCreateInteractor.swift]: Yeah that's probably a more proactive strategy. Will update.

**Comment by woudini** [comment]
Currently not receiving conversion socket messages even though the app is subscribing and receiving other socket messages.

**Comment by bchrisarriola** [comment]
Looks like conversions are coming through dev though. Will test on that env.

**Pull request merged**